### ISSUE TITLE
Feedback on MML vs SSW - Only here for safe keeping.

### ISSUE THREAD

**Issue by HammerGS** [MegaMek/megameklab#37]
Received some feedback on MML vs SSW on the forums.  Placing it here so it doesn't get lost.  Just for reference.
## \- SSW allows me to add a picture of the Mech, and a logo, and Pilot info, which makes it easier for newer players to figure out what 'Mech is what on the field and what their skills are.
- Gave up trying to figure out how to add Experimental Equipment like Null Sig/Void Sig etc. Stuff that is on the basic setup tab on SSW. Or stuff that should be easy to find, like Artemis IV and CASE. I have to load a 'Mech in MML from an appropriate year or source/era before I get access to tech that I should be able to see just by switching the Tech Level drop-down box.

**Comment by HammerGS** [comment]
Additional Feedback from Reddit - https://www.reddit.com/r/battletech/comments/5gl1fo/best_mechlab_for_tabletop_use/

Relevant stuff:

There's a bunch of little friction points that make me hate MML.

For example, add ons to weapons, such as Artemis or PPC Capacitors. In SSW, it's as simple as hitting a check box (for Artemis, Apollo, etc) or opening a contextual menu (for PPC Caps, Laser Insulators, etc). In MML, you have to add those individually as separate components.

SSW also makes it very easy to design OmniMechs, using a simple checkbox and a handful of buttons to manage the variants. To this day, I have yet to figure out how to make an Omni in MML.

It's easy in SSW to quickly allocate bulk items to slots, such as Endo Steel or a pile of lasers. In MML, I have to drag each item to the slot I want.

**Comment by Tamren** [comment]
Agreed, for mech work SSW is quite excellent. I would not mind it at all if MML just shamelessly copied it's UI and allowed us to apply it to vehicles and infantry as well.

**Comment by HammerGS** [comment]
Another comment today on FB...re MML v SSW

> I use Solaris Skunk Werks unless it's a very new unit that requires MegaMekLab. I hope someday MML has the option to print with advanced tables on the side.

**Comment by Xeno426** [comment]
One aspect I'd love to see would be the manufacturing info that SSW has, along with the tabs for history, deployment, etc added to MML. It's nice having the mech file of a custom design contain the fluff required for it.

Oh, and UMU. You can actually add those to mechs in SSW without opening the mech file and manually editing them in.

**Comment by neoancient** [comment]
UMU is working after my partial ui overhaul. I agree about text fields for fluff data.

**Comment by mhjacks** [comment]
More feedback from my group today:  +1 for printing the TW tables (like SSW does) as an option in MML.  MML is great because it includes 3140-era units (which we played with today),

**Comment by FrozenIceman01** [comment]
Agreed on the TW tables.  That is definitely something that we have definitely missed in my group as well.  Also an option to print the Mech weapon to anti-infantry damage would be great as well.

**Comment by arlith** [comment]
That's something I've wanted to add for a while, but haven't devoted the time to it.  MML used to print them, but then it switched to print quality sheets using SVGs.  At that time, it wasn't supported, but I've been working on an idea to separate the sheet layout from the data that gets printed, so users can edit their own SVG files and print to those.  I've got this mostly working for warship printing, but haven't had much time to finish it.  Someday, I'd like to take the ideas  there and apply them to mech sheets.

**Comment by Xeno426** [comment]
One thing I've noticed is that while SSW will crop or scale the mech/vehicle/etc images when a unit sheet is printed, MML will always stretch. IMO, the scaling looks much better than the stretching that MML tries to do.

**Comment by MGM175** [comment]
I was thinking to post a issue, where you could save or (more useful) read SSW file to/from Solaris Skunk Werks, but right now I'll halt with that.

Is there any plans to actually **merge** SSW and MML?
Of course, I would think that it would work due to different licences, but asking wouldn't hurt. ;)

**Comment by HammerGS** [comment]
@MGM175 - Sorry, never going to happen for a variety of reasons.

**Comment by HammerGS** [comment]
https://github.com/MegaMek/megameklab/issues/176#issuecomment-414782446

**Comment by HammerGS** [comment]
See also https://github.com/MegaMek/megameklab/issues/144

**Comment by HammerGS** [comment]
Another piece of feedback comparing SSW to MML

https://bg.battletech.com/forums/index.php?topic=63414.msg1458022#msg1458022

**Comment by HammerGS** [comment]
Interesting Feedback from a FB post on RS.

"Perhaps making the Sheet landscape, shrinking the crit locations, increase the Structure/Armor overlay (basically make the silhouette larger and the dots better spread out, charts to speed play, things like that is all."

**Comment by Drudenhaus** [comment]
The lack of quick reference charts in record sheets is a blocker for me from fully migrating from SSW to MML. I only use MML for things not supported in SSW, such as VSP lasers. It's been more than a year since #176 was opened and there doesn't seem to be much traction on including this option. Is this a person-hours issue or something else?

**Comment by neoancient** [comment]
It's definitely a time thing. It's something I've started on, but so many other things pop up and demand attention. I have a record sheet overhaul potentially looming in the near future and may get to it then, but I can't guarantee anything.

**Comment by Drudenhaus** [comment]
Interesting. Raises a few follow up questions:
1. Do you think this would be a suitable first contribution? The PrintRecordSheet class looks fairly large and probably only one of a few components that needs modification.
2. If yes, do you have a design you're working from?

**Comment by neoancient** [comment]
It wouldn't be very complicated, but would require a good working knowledge of SVG and DOM. The record sheet generation doesn't need to change at all. It's just a matter of creating an SVG document in memory, then embedding the charts and currently generated record sheet (scaled down) in it, then printing that instead. Most of the work would involve creating the tables in SVG and figuring out where everything fits. Even without the more detailed technical skills just planning the layout or producing SVG versions of the tables would be a big help.

**Comment by FrozenIceman01** [comment]
Thank you all for looking back into this, I know it has been awhile but I
definitely think that it would go a long way to increasing usability.

I would like to help with this as well, if Sean and you are willing.  I do
have some graphics experience and I believe GIMP may have a plugin to
create SVG files.  I can look into possible layouts and mockups as well to
help as well.

Thank you,
Andrew

**Comment by HammerGS** [comment]
Some additional thoughts on a cheater chart from the forums - 
https://bg.battletech.com/forums/index.php?topic=66839.0

**Comment by rjhancock** [comment]
To stay in compliance with CGL, maybe an optional second page with the hit tables and charts like that?

**Comment by Drudenhaus** [comment]
> To stay in compliance with CGL, maybe an optional second page with the hit tables and charts like that?

1. Why would adding an optional feature be a "compliance" issue?
2. A second page adds basically no benefit. Reference sheets already exist (in official form, even). Adding the tables on the sheet is a huge benefit for anybody that doesn't have them committed to memory.

**Comment by rjhancock** [comment]
Since CGL uses MML for their official record sheets, it's best not to upset those that could shut down a project for copyright infringement.

And as a second page, it allows for less clutter when viewing.

**Comment by Drudenhaus** [comment]
> Since CGL uses MML for their official record sheets, it's best not to upset those that could shut down a project for copyright infringement.


I still don't understand. The reference tables in SSW are optional (you can turn them on and off). 
This would have literally zero impact to CGL if the MML defaulted to reference tables = off.

**Comment by neoancient** [comment]
I removed a useless argument about a non-issue.

**Comment by plutonick** [comment]
I'd like to comment that I too wish for a tables and charts thing option, like SSW uses. Being an option it wouldn't impact CGL using the software for official products.


I was going to post a feature request, but I see there are already two similar request, so I just comment here.

**Comment by WEKarnesky** [comment]
As a SSW dev, I just want to share that I love what MML does too, and I personally don't view that they need to compete.  I see them as complementary products with slightly different ideas and goals despite their significant overlap.  

It would be awesome if Megamek tried to be compatible with SSW, but we are also doing our best to be compatible with MML.  We already have (and have had) outputting to MML's save format, and we are working on loading MegaMek files.

**Comment by neoancient** [comment]
I agree. Since SSW wasn't in active development during the time I've been on the MM project I've never concerned myself with the SSW loader. It's something that needs to be dusted off and brought up to date.

**Comment by MGM175** [comment]
@WEKarnesky Allright, but let's discuss a  possible problem. What if SSW gets a ability to read MML files, but contains, let's just say - Dark Age tech, not supported by SSW?

**Comment by WEKarnesky** [comment]
@MGM175 We will have to error out of the load gracefully or do a partial load, depending on what is missing (for example simple equipment like weapons we could not load and leave the hole open to be replaced).  This is the same thing we'd have to do for any non-canon equipment that MML has, like the RAC/20; or any equipment that a user just types into MML's save file..

   That said, we are also currently working on adding any missing mech equipment and welcome any tickets.

**Comment by SJuliez** [comment]
Additional: https://bg.battletech.com/forums/index.php?topic=67812.msg1569133#msg1569133

**Comment by johnboyjjb** [comment]
I'm surprised nobody is asking for a chart like this:
![Screenshot 2024-05-15 170053](https://github.com/MegaMek/megameklab/assets/75278541/47642628-d80f-446f-9d86-8ed888a384e3)
Or personally, even better, like this:
![Screenshot 2024-05-15 170207](https://github.com/MegaMek/megameklab/assets/75278541/428e7138-d2a7-468f-8156-062281a5b08b)
